### PR TITLE
BUG: conversion of Series to Categorical

### LINKED ISSUE #16524: Converting Series to categorical
When converting `Series` to `categorical` dtype, the following method

```python

pd.Series(pd.Series(['a', 'b', 'c']), dtype='category')

```

fails with  traceback

```pytb
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-32-6c903a3f0db3> in <module>()
----> 1 pd.Series(pd.Series(['a', 'b', 'c']), dtype='category')

/Users/ttylec/.virtualenvs/zettafox/lib/python3.6/site-packages/pandas/core/series.py in __init__(self, data, index, dtype, name, copy, fastpath)
    241                 if dtype is not None:
    242                     data = data.astype(dtype=dtype, raise_on_error=False,
--> 243                                        copy=copy)
    244                 elif copy:
    245                     data = data.copy()

/Users/ttylec/.virtualenvs/zettafox/lib/python3.6/site-packages/pandas/core/internals.py in astype(self, dtype, **kwargs)
   3222 
   3223     def astype(self, dtype, **kwargs):
-> 3224         return self.apply('astype', dtype=dtype, **kwargs)
   3225 
   3226     def convert(self, **kwargs):

/Users/ttylec/.virtualenvs/zettafox/lib/python3.6/site-packages/pandas/core/internals.py in apply(self, f, axes, filter, do_integrity_check, consolidate, **kwargs)
   3089 
   3090             kwargs['mgr'] = self
-> 3091             applied = getattr(b, f)(**kwargs)
   3092             result_blocks = _extend_blocks(applied, result_blocks)
   3093 

/Users/ttylec/.virtualenvs/zettafox/lib/python3.6/site-packages/pandas/core/internals.py in astype(self, dtype, copy, errors, values, **kwargs)
    469     def astype(self, dtype, copy=False, errors='raise', values=None, **kwargs):
    470         return self._astype(dtype, copy=copy, errors=errors, values=values,
--> 471                             **kwargs)
    472 
    473     def _astype(self, dtype, copy=False, errors='raise', values=None,

/Users/ttylec/.virtualenvs/zettafox/lib/python3.6/site-packages/pandas/core/internals.py in _astype(self, dtype, copy, errors, values, klass, mgr, **kwargs)
    488         # this is only called for non-categoricals
    489         if self.is_categorical_astype(dtype):
--> 490             return self.make_block(Categorical(self.values, **kwargs))
    491 
    492         # astype processing

TypeError: __init__() got an unexpected keyword argument 'raise_on_error'
```

When using simple list or np.array instead of pd.Series, it work. Doing `.astype('category')` also works.

<details>
INSTALLED VERSIONS
------------------
commit: None
python: 3.6.1.final.0
python-bits: 64
OS: Darwin
OS-release: 15.6.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: en_US.UTF-8
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.20.1
pytest: None
pip: 9.0.1
setuptools: 35.0.1
Cython: 0.25.2
numpy: 1.12.1
scipy: 0.19.0
xarray: None
IPython: 5.3.0
sphinx: None
patsy: None
dateutil: 2.6.0
pytz: 2017.2
blosc: None
bottleneck: None
tables: 3.3.0
numexpr: 2.6.2
feather: None
matplotlib: 2.0.0
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: 0.999999999
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.9.6
s3fs: None
pandas_gbq: None
pandas_datareader: None
</details>

### PR DESCRIPTION
**Pull request by preddy5**
Patch to bug fix #16524

### CODE REVIEW

**Comment by preddy5** [comment]
@jreback this fixes the issue, could you review the PR and let me know if am doing it wrong.

**Comment by jreback** [comment]
pls add tests. in fact you want to write the tests first to make sure that they fail, then your fix makes it pass. also add a whatsnew entry (0.20.2) is fine.

File: pandas/tests/series/test_dtypes.py [hunk 1]
```diff
@@ -248,3 +249,11 @@ def test_intercept_astype_object(self):
 
         result = df.values.squeeze()
         assert (result[:, 0] == expected.values).all()
+
+    def test_series_to_categorical(self):
+        series = Series(['a', 'b', 'c'])
+
+        categorical_series = Series(series, dtype='category')
+        expected = Categorical(['a', 'b', 'c'])
```
Comment (TomAugspurger) [inline pandas/tests/series/test_dtypes.py]: I think you can mak this `expected = Series(['a', 'b', 'c'], dtype='category')`

and then you can use `tm.assert_series_equal(categorical_series, expected)`.

**Comment by preddy5** [comment]
@TomAugspurger  can you retrigger the travis build?

**Comment by preddy5** [comment]
could you retrigger circleci build as well please. Thank you.

**Comment by preddy5** [comment]
@jreback @TomAugspurger Could you review the PR please, Thank you.

File: doc/source/whatsnew/v0.20.2.txt [hunk 2]
```diff
@@ -59,6 +59,7 @@ Conversion
 
 - Bug in :func:`to_numeric` in which empty data inputs were causing a segfault of the interpreter (:issue:`16302`)
 - Silence numpy warnings when broadcasting ``DataFrame`` to ``Series`` with comparison ops (:issue:`16378`, :issue:`16306`)
+- Bug in ``Series`` with dtype='category' (:issue:`16524`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.20.2.txt]: Bug in ``Series`` construction with ``dtype='category'``

File: pandas/tests/series/test_dtypes.py [hunk 3]
```diff
@@ -248,3 +248,11 @@ def test_intercept_astype_object(self):
 
         result = df.values.squeeze()
         assert (result[:, 0] == expected.values).all()
+
+    def test_series_to_categorical(self):
+        series = Series(['a', 'b', 'c'])
```
Comment (jreback) [inline pandas/tests/series/test_dtypes.py]: add comment for the issue number

File: pandas/tests/series/test_dtypes.py [hunk 4]
```diff
@@ -248,3 +248,11 @@ def test_intercept_astype_object(self):
...
+
+        categorical_series = Series(series, dtype='category')
```
Comment (jreback) [inline pandas/tests/series/test_dtypes.py]: result =

File: pandas/tests/series/test_dtypes.py [hunk 5]
```diff
@@ -248,3 +248,11 @@ def test_intercept_astype_object(self):
...
+        expected = Series(['a', 'b', 'c'], dtype='category')
+
+        tm.assert_series_equal(categorical_series, expected)
```
Comment (jreback) [inline pandas/tests/series/test_dtypes.py]: categorical_series -> result

File: pandas/core/categorical.py [hunk 6]
```diff
@@ -248,7 +248,8 @@ class Categorical(PandasObject):
     __array_priority__ = 1000
     _typ = 'categorical'
 
-    def __init__(self, values, categories=None, ordered=False, fastpath=False):
+    def __init__(self, values, categories=None, ordered=False, fastpath=False,
```
Comment (jreback) [inline pandas/core/categorical.py]: this fix should not be here.

instead in core/internals.py/Block/_astype.py

``raise_one_error`` can be a named argument in the signature and not passed to the constructor

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by preddy5** [comment]
@jreback I have made the changes according to your comments.

**Comment by preddy5** [comment]
@jreback hey I have made changes according to your previous comments, could you review the changes once. Thank You.

**Comment by TomAugspurger** [comment]
@preddy5 I rebased and moved the release note to 0.20.3 (releasing in a couple weeks).

I'll let Jeff comment on the changes, but I think we're good.

File: doc/source/whatsnew/v0.20.3.txt [hunk 7]
```diff
@@ -44,6 +44,7 @@ Conversion
 ^^^^^^^^^^
 
 - Bug in pickle compat prior to the v0.20.x series, when ``UTC`` is a timezone in a Series/DataFrame/Index (:issue:`16608`)
+- Bug in ``Series`` with ``dtype='category'`` (:issue:`16524`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.20.3.txt]: be a bit more descriptive.

``Bug in Series construction when passing type=category``

**Review by jreback — APPROVED** [review]
tiny change. ping on green.

**Comment by jorisvandenbossche** [comment]
@preddy5 Thanks!

**Pull request merged**